fea wip

NajlaAbassi opened the pull request:
establishing the implementation of fea slot:

- [x] constructor function takes enrichment results as a named list of topGO(data.frame)/enrichResults objects
- [x] attempts to link FE results to existing DE results in dde object
- [x] methods to fetch, add, and remove FE results from dde object
- [x] fea() returns a shaked version of the results, to be easily plugged into a GTL later
- [x] summary() method
- [x] handle the shaking inside fea correctly for different outputs
- [x] update example data to reduce size
- [x] add custom shakers
- [x] validate object
- [x] update test unit
- [x] vignette

NajlaAbassi committed:
defined key sections in vignette worked on the initial content of it

NajlaAbassi committed:
fea method, fetches the shaked result table, if not available returns the original object

NajlaAbassi committed:
updating add_fea method

NajlaAbassi committed:
update remove_fea method

NajlaAbassi committed:
updating constructor function, as well as helper functions for checking the fea input validity, type, as well as adding extra checks

NajlaAbassi committed:
updating news + version

NajlaAbassi committed:
update dependencies

NajlaAbassi committed:
adding toy enrich res

NajlaAbassi committed:
update test unit

NajlaAbassi committed:
add FEA data

NajlaAbassi committed:
move all internal functions into a separate file

NajlaAbassi committed:
update the constructor function to accept know FE results from david, fgsea, gsea, enrichr and gprofiler, as well as use corresponding shakers (from GeneTonic or internal custom ones)

NajlaAbassi committed:
update summary, and the add_fea method to accept new input types, and add a new param, force, to overwrite results

NajlaAbassi committed:
update validity function

NajlaAbassi committed:
update documentation

NajlaAbassi committed:
update tests round 1 :v

NajlaAbassi committed:
leaving this suggestion (inheritance from sce) here for now

NajlaAbassi committed:
update fea_type

NajlaAbassi committed:
update example data, to reduce package size

NajlaAbassi committed:
remove GeneTonic from dependencies, and add internal custom shakers

NajlaAbassi committed:
update methods and documentation

NajlaAbassi committed:
update tests

NajlaAbassi committed:
documenting data!

NajlaAbassi committed:
stricter condition to break the function exactly when it should stop

NajlaAbassi committed:
adding more example data + documenting it

NajlaAbassi committed:
adjusting the column names for gProfiler and gsea, as well as enhance error checks

NajlaAbassi committed:
checks before everythning!

NajlaAbassi committed:
updating test units- round 2

NajlaAbassi committed:
setup data for unit test

NajlaAbassi committed:
update readme with relevant info

NajlaAbassi committed:
update readme

NajlaAbassi committed:
remove unused data

federicomarini commented:
Very well done so far!
I have a couple of comments that are arising in my "test drive", I will pass them on through mattermost 😉

NajlaAbassi committed:
add new methods to rename DEAs and FEAs

NajlaAbassi committed:
updating version and news

NajlaAbassi committed:
updating documentation

NajlaAbassi committed:
updating methods dea and fea to throw explicit and comprehensive warning if dea/fea_name is not provided by the user, and returns the 1st entry + improve error msgs

NajlaAbassi committed:
adding missing examples

NajlaAbassi committed:
add the option to overwrite DEA results in case of name overlap

NajlaAbassi committed:
fixing typo in param name!! thanks fede for spotting that!

NajlaAbassi committed:
removing unecessary checking

NajlaAbassi committed:
adding tests, round 3

NajlaAbassi committed:
update readme

NajlaAbassi committed:
adding get_fea_list, to fetch FEA results as a list, and if the dea_name is specified, return only the associeted FEAs

NajlaAbassi committed:
adding assign_dea_to_fea method, a better way to link *manually* FEAs to DEAs (especially if name conventions are not followed)

NajlaAbassi committed:
move david output into extdata folder

NajlaAbassi committed:
adding example

NajlaAbassi committed:
updating the "getters" dea() and fea(),by adding the param format, so that user can choose whether to fetch a minimal/standard version of the results or the original object

NajlaAbassi committed:
update documentation

NajlaAbassi committed:
adding tests

NajlaAbassi committed:
update news

NajlaAbassi committed:
adding pkgs that the vignette depends on

NajlaAbassi committed:
update the dea() method to return extra columns from the rowData

NajlaAbassi committed:
update documentation

NajlaAbassi committed:
updating the signature for all the methods

NajlaAbassi committed:
adding the necessary checks for all arguments

NajlaAbassi committed:
adding a new arg for dea, to return extra columns from the rowData

NajlaAbassi committed:
update fea, since i forgot to use the de_name argument

NajlaAbassi committed:
simplfy condition

NajlaAbassi committed:
adding extra_rd param to add extra coloumn in the minimal version of dea

NajlaAbassi committed:
updated man

NajlaAbassi committed:
update tests

federicomarini committed:
catching the names of the objects provided + removing unessential columns in get_dea_list's output

federicomarini committed:
Merge branch 'fea_wip' of github.com:imbeimainz/DeeDeeExperiment into fea_wip

federicomarini committed:
adding proper sectioning to the methods source file, to simplify navigation via IDE outline

federicomarini committed:
no need to remove the columns anymore as this is now handled in the `dea()` function

federicomarini committed:
fixing get_fea_list to work also when a specific dea_name is defined as a parameter

NajlaAbassi committed:
adding method add_scenario_info, to attach extra info about a specific DE contrast

NajlaAbassi committed:
updating dea_rename, so that also the de_name for linking dea and fea is also renamed!

NajlaAbassi committed:
updating documentation

NajlaAbassi committed:
adding tests

federicomarini committed:
adding URL and BugReports fields to the DESCRIPTION

NajlaAbassi committed:
this will anyway create a conflict :v

NajlaAbassi committed:
Merge branch 'fea_wip' of github.com:imbeimainz/DeeDeeExperiment into fea_wip

federicomarini committed:
for consistency, Contrast becomes DEA_name in the summary function

NajlaAbassi committed:
rename method "assign_dea_to_fea" to "link_dea_and_fea"

NajlaAbassi committed:
update summary to: handle the clusterpro objs, and display the scenario_info

NajlaAbassi committed:
update remove_dea, to optionally remove linked feas

NajlaAbassi committed:
dont be lazy and called clusterProfiler :v

NajlaAbassi committed:
fea_tool instead of dea_type

NajlaAbassi committed:
update namespace and some returned msgs (still testing the cli pkg)

NajlaAbassi committed:
resolve conflict

Merge branch 'fea_wip' of github.com:imbeimainz/DeeDeeExperiment into fea_wip

# Conflicts:
#	R/DeeDeeExperiment-methods.R

NajlaAbassi committed:
fix name fea_tool!

NajlaAbassi committed:
fix wrong arg assignment

NajlaAbassi committed:
handle correctly clusterpro res in summary

NajlaAbassi committed:
fea_tool param in add_fea can take more tools if fea is a list of more than 1 element

NajlaAbassi committed:
fixing the fact that fe_name was never properly used :v

NajlaAbassi committed:
update validity error msg

NajlaAbassi committed:
add stringr in the dependencies

NajlaAbassi committed:
updating msgs with cli

NajlaAbassi committed:
set verbose to false by default

NajlaAbassi committed:
adding a new line so that warnings dont print in the same line as the show output

NajlaAbassi committed:
updating tests

NajlaAbassi committed:
increasing the test coverage to ~90% 🙌

NajlaAbassi committed:
check not needed

NajlaAbassi committed:
loading more test data

federicomarini committed:
moving the first check a bit up to get an informative message thrown if no dea_name is provided

federicomarini committed:
adding a tiny test for the scenario info in the summary

federicomarini committed:
drafting a small test to check we cover the proper working of the david output

federicomarini committed:
hm, a temporary fix to check the GHA in action..

federicomarini committed:
removing a bit of suggested packages to alleviate the dependency load especially if they won't really be used in the final form of the package..

NajlaAbassi committed:
adding the vignette for the first round of tests

NajlaAbassi committed:
update readme

NajlaAbassi committed:
update msg format

federicomarini committed:
the anatomy file is already at the same level, so this prefix can be removed

federicomarini committed:
avoiding the need of having ideal and BIUMmisc as explicit dependencies, even if just suggested

federicomarini committed:
inverting the order of what needs to be checked for the enrichr deedeefication

federicomarini committed:
only accept the expected format for the DE elements

federicomarini committed:
having David fail a bit more

federicomarini committed:
some extra triggered fails for fgsea and enrichr

federicomarini committed:
adding back the summary+info test, works in my hands somehow (can be commented out if needed)

federicomarini commented:
I have added a couple of extra tests with a small fix to a check on the columns.

Works well in my hands too - I was just considering the option to stay away from the whole `resuSet` and possibly stick to the "classical ways" of generating such files. Yes, possibly not too efficient. But I think this would be the way that most possible users could relate to, as they will come with their objects&results "anyways".

I will have a good look at the whole thing, with some comments here'n'there as an official github review ;)

federicomarini committed:
reducing the dependency load a bit by removing stringr as a whole (always heavy in being compiled, and we need not too much in terms of "performance", so we can use base strwrap+paste)

federicomarini committed:
adding apeglm as dependency, as suggested by Ludwig (gets an error in the vignette otherwise..)

federicomarini committed:
this workaround should give us the possibility to "properly define" the summary method, in a way that the arguments are detected

federicomarini committed:
Depending rather than importing SummarizedExperiment. Why? So that you can interact with a DeeDeeExperiment object along the usual lines for interacting with SE-derivates from just loading your package via library --> **all** SE functions are readily available for objects of your DDE class

federicomarini committed:
Version 0.5.0, given the full set of changes...
And well, removing the Remotes field as it cannot solve our issues anyways (need the fix for topGO on main Bioc)

federicomarini committed:
while at it, adding dependency on a more recent version of R (also for compliance with Bioc guidelines)

federicomarini committed:
some more precise namespace import specs

federicomarini committed:
avoiding paste in stop() call as per BiocCheck note

federicomarini committed:
adding chunk labels for all the vignette components

federicomarini committed:
excluding WIP vignette file (if at all present)

federicomarini committed:
specifying the @format field, and while we are at it, adding some nice combo of seealso with the @family tag

federicomarini commented on R/DeeDeeExperiment-methods.R:
```diff
@@ -651,78 +1429,138 @@ setMethod("show",
+          lfc <- rowData(object)[[lfc_col]]
+          padj <- rowData(object)[[padj_col]]
+          sum(lfc < 0 & padj < FDR, na.rm = TRUE)
+        } else {
```
Can we switch to the vapply implementation?
There is a note in BiocCheck that we can avoid triggering.

I am marking here two spots, but we can do this for the whole set of spots where we use in the summary the `sapply`.

federicomarini commented on DESCRIPTION:
```diff
@@ -18,26 +18,42 @@ Authors@R:
 Encoding: UTF-8
 Roxygen: list(markdown = TRUE)
 RoxygenNote: 7.3.2
+Depends:
+    R (>= 4.5.0),
+    SummarizedExperiment
```
I ended up moving it here after a quick conversation with Ludwig, happy to share more detail offline 😉 
tl;dr: we should simply have it all "ready to use" and this is the simple way to get it done like this!

federicomarini commented on DESCRIPTION:
```diff
@@ -18,26 +18,42 @@ Authors@R:
     S4Vectors,
     utils,
     DESeq2,
     edgeR,
-    limma
+    limma,
+    cli
 Suggests: 
     macrophage,
     knitr,
     BiocStyle,
+    apeglm,
+    mosdef,
+    DT,
+    org.Hs.eg.db,
+    biomaRt,
+    topGO,
+    clusterProfiler,
+    GeneTonic,
+    ReactomePA,
+    pcaExplorer,
+    enrichplot,
```
we should consider having most deps if at all needed as Suggests: entries. 
This would keep the stack as slim as possible!

On a related note: We can (and should) not do too much in terms of analysis and further steps, as this would also inflate it a bit.

federicomarini commented on DESCRIPTION:
```diff
@@ -1,6 +1,6 @@
 Package: DeeDeeExperiment
-Title: DeeDeeExperiment: An S4 Class for Managing Transcriptomic Analysis results
-Version: 0.2.0
+Title: DeeDeeExperiment: An S4 Class for Managing and exploring omics Analysis results
```
do we need it in full Title Case? Exploring, Omics, Results would be needed?

NajlaAbassi replied:
true! that was the very first version of the title and should be updated

federicomarini committed:
updated news

federicomarini commented on tests/testthat/test-DeeDeeExperiment.R:
```diff
@@ -1,4 +1,6 @@
 test_that("creating", {
```
I am placing the comment here just to have it somewhere.
A thought - maybe worth a lot to improve the readability of all this. Or maybe two..

1. we could use sectioning of this one file (the `# text -------` trick, with the RStudio shortcut?)
2. we could split this one megafile into smaller ones based on their focus of functionality being tested

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
+```{r loadlibraries}
+library("DeeDeeExperiment")
+
+library("macrophage")
+library("DESeq2")
+library("edgeR")
+library("mosdef")
+library("GeneTonic")
+library("org.Hs.eg.db")
+library("topGO")
+library("clusterProfiler")
+library("ReactomePA")
+library("pcaExplorer")
````
I am thinking we can and should cut this down to the essential.
This will keep us slim, but also does not force users to make a too wide "thought jump" to understand why we use this.
Dependencies being in the Suggests: are already a good thing if (as we do) we do not import any of their functionality per se. I don't know exactly how the class types are handled, but we can try to cut down and potentially put back again if needed...

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
+
+```{r alltheresustep, cache=TRUE}
+# DE with DESeq2
+alltheresults <- function(resuSet, 
+                          dds_obj, 
+                          contrast, 
+                          FDR,
+                          logFC_threshold,
+                          anno_df, 
+                          anns, 
+                          species) {
+  
+  id_contrast <- contrast
+  resuSet[[id_contrast]] <- list()
+  
````
This is probably most interesting to us internally, but I think outside of our circle noone would probably use this but rather "their own resuSet way" - or simply, and this is perfectly fine, "just a large set of objects".

Actually the point of "showing that we would carry along a large set of objects" is one of the key motivations behind DeeDeeExp as a class.

Can we possibly rewrite this one part, in a way that (at the cost of sounding boring) does this without too much extras (no resuSet, no MA plot, and actually without even needing the enhanced DT version... - all these things *could* potentially come later again, but probably once into DeeDeeExp)

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
+```{r limmavoomde}
+# DE with limma-voom
+
+limma_results <- function(resuSet,
+                          MArrayLM_obj,
+                          contrast,
+                          FDR,
+                          logFC_threshold,
+                          anno_df,
+                          anns,
+                          species) {
````
in a possibly similar fashion, also this one is not too much needed. Unless, well, we make a solid point of "a simple loop iteration", but that one can really be an essential one, i.e. without anno_df and all other "embellishments" which, for sure, are extremely useful but not essential.
Deal?

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
+```
+
+
+# DeeDee in the DeeDee way
+
````
This section might understandably deserve a bit more of corollary text to show&tell a bit more what goes on, but this is perfectly fine to keep it like this for now.

We should capitalize on the feedback of the others as well to see if they spot something here missing and we should add that plus their textual explanations...

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
-```{r cite}
-citation("DeeDeeExperiment")
+# get me all feas for this de_name
+lapply(DeeDeeExperiment::get_fea_list(dde_macs, 
+                                      dea_name = "myresuset_macs$condition_IFNg_vs_naive$res_DESeq",
+                                      format = "original"),
+       head) # as expected, no linked de , so nothing is returned
````
this is admittedly only to not take up too much output in the vignette. We can actually do it on one single list element tbh

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
+# fea_info<-   ## same
+## or again: do we even need the SETTERS for these?
+
+```
+
````
We probably could make good use here of a section on how to use this within a single-cell dataset context.
This might again increase the number of dependencies we need to at least Suggest.
Alternatively, and this is probably an even better way of conceptually showing this one, this can be a separate vignette - imagining the title being "Using DeeDeeExperiment with single-cell RNA-seq data". The whole motivation part is kinda clear from the main vignette, but this is clearly (at least to me) a different use case, where `dde` can do a little bit of wonders.
Then again - we will have to see if the "plug of the SCE object" inside this will... just work, or if we need to tweak a bit of things.
Oooooor: We opt to build `dde` upon `sce` rather than on `se`.

federicomarini commented on vignettes/DeeDeeExperiment_manual.Rmd:
````diff
@@ -92,33 +124,627 @@ Once installed, the package can be loaded and attached to your current workspace
+
+# done, i guess? TODO; handle the cases where we could or should be more or less verbose
+
+# TODO: a good plug and use case to justify using this in SingleCellExperiment landia
 ```
````
see the comment above on this, probably best suited in a separate vignette?

A clever thing to do to would then be to have the VignetteIndexEntry field set up in a way that things (= vignetteS) will be displayed in the right "sequence".

Example:

<img width="397" alt="image" src="https://github.com/user-attachments/assets/8e1756e8-8084-432d-85b0-0a022388233e" />

so that they look like...

<img width="519" alt="image" src="https://github.com/user-attachments/assets/1c21d4e9-aeeb-4941-8142-ee9bb1db8218" />
as on the Bioc landing page.

federicomarini commented on vignettes/myresuset_macs.RData:
I am not too sure we need this one ultralarge thing in.
I mean, it is fantastic to have it ready for testing but maybe we can simply "just not have it in anymore" under git version control, as a starter, and also as a "not to be even considered when building" --> add into .Rbuildignore

NajlaAbassi committed:
adding warnings for mismatched genes when using add_dea + update the warning message to show that mismatches are not fatal, but have consequences that the user should be aware of

federicomarini committed:
this should save us about 7-8 Mb of "pure textual content" that would really not help the flow of the vignette

federicomarini committed:
remove from git tracking the myresuset object

federicomarini committed:
avoid saving/using in the builds the myresuset Rdata object

federicomarini committed:
better spec for the RData file in the .Rbuildignored  files

federicomarini committed:
Merge branch 'fea_wip' of github.com:imbeimainz/DeeDeeExperiment into fea_wip

federicomarini reviewed (commented):
Okiedokie, "full official round" of review is done.
I think we are extremely well en route 😉 

Have a look at some of the proposed changes, and let's finish working on the vignette(s? see the comment on the single cell data related one - we could also keep it for a later point)

NajlaAbassi committed:
adding a new function supported_fea_formats() to get the possible fea input formats right away (and avoid long error msgs)

NajlaAbassi committed:
Merge branch 'fea_wip' of github.com:imbeimainz/DeeDeeExperiment into fea_wip

NajlaAbassi committed:
update error msg to be shorter and encourage the use of supported_fea_formats() to get the whole list of currently accepted tools

NajlaAbassi committed:
structuring tests into differents files based on functionality

NajlaAbassi committed:
using vapply instead of sapply

NajlaAbassi committed:
ignore vignette cache
